<launch>

    <param name="use_sim_time" value="false"/>
    
    <!-- lidar operation -->
    <include file="$(find rplidar_ros)/launch/two_lidars.launch"/>

    <!-- front lidar -->
    <node pkg="sos_fusion" type="clustering" name="clustering_node"/>
    <node pkg="sos_fusion" type="fusion" name="fusion_node"/>

    <!-- behind lidar -->
    <node pkg="sos_fusion" type="clustering_back" name="clustering_back_node"/>

    <!-- camera find person -->
    <include file="$(find camera_degree)/launch/camera.launch"/>

    
    <!-- cart main -->
    <node pkg="ros_servo" type="ros_control" name="ros_control_node"/>

    <node pkg="ros_servo" type="ros_button" name="ros_button_node"/>


    <!-- xbox joystick -->
    <node name="xbox_joystick" pkg="joy" type="joy_node"/>

    <!-- arduino cart control -->
    <node name="serial_node_1" pkg="rosserial_python" type="serial_node.py">
    <param name="port" value="/dev/arduino_mega"/>
    <param name="baud" value="57600"/>
    </node>

    <!-- arduino cart button -->
    <node name="serial_node_2" pkg="rosserial_python" type="serial_node.py">
    <param name="port" value="/dev/arduino_uno"/>
    <param name="baud" value="57600"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find ros_servo)/rviz/run_all_temp.rviz"/>

</launch>
